src/impl/realm_coordinator.cpp

```diff
@@ -348,9 +351,9 @@ void RealmCoordinator::commit_write(Realm& realm)
         m_notifier->notify_others();
     }
 #if REALM_ENABLE_SYNC
-    if (m_sync_session) {
+    if (m_sync_session && !realm.is_closed()) {
```

Comment:
Yup, you are right . Moved. @austinzheng would you please take a look of my last commit?